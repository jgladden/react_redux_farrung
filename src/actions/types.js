export const SET_SECTION = 'SET_SECTION';

export const GET_PORTFOLIO_ITEMS = 'GET_PORTFOLIO_ITEMS';
export const GET_PORTFOLIO_ITEMS_SUCCESS = 'GET_PORTFOLIO_ITEMS_SUCCESS';
export const GET_PORTFOLIO_ITEMS_ERROR = 'GET_PORTFOLIO_ITEMS_ERROR';

export const EDIT_PORTFOLIO_ITEM = 'EDIT_PORTFOLIO_ITEM';
export const EDIT_PORTFOLIO_ITEM_SUCCESS = 'EDIT_PORTFOLIO_ITEM_SUCCESS';
export const EDIT_PORTFOLIO_ITEM_ERROR = 'EDIT_PORTFOLIO_ITEM_ERROR';

export const ADD_PORTFOLIO_ITEM = 'ADD_PORTFOLIO_ITEM';
export const ADD_PORTFOLIO_ITEM_SUCCESS = 'ADD_PORTFOLIO_ITEM_SUCCESS';
export const ADD_PORTFOLIO_ITEM_ERROR = 'ADD_PORTFOLIO_ITEM_ERROR';

export const REMOVE_PORTFOLIO_ITEM = 'REMOVE_PORTFOLIO_ITEM';
export const REMOVE_PORTFOLIO_ITEM_SUCCESS = 'REMOVE_PORTFOLIO_ITEM_SUCCESS';
export const REMOVE_PORTFOLIO_ITEM_ERROR = 'REMOVE_PORTFOLIO_ITEM_ERROR';

export const POST_AUTH = 'POST_AUTH';
export const POST_AUTH_SUCCESS = 'POST_AUTH_SUCCESS';
export const POST_AUTH_ERROR = 'POST_AUTH_ERROR';
export const AUTH_LOGOUT = 'AUTH_LOGOUT';

export const POST_MESSAGE = 'POST_MESSAGE';
export const POST_MESSAGE_SUCCESS = 'POST_MESSAGE_SUCCESS';
export const POST_MESSAGE_ERROR = 'POST_MESSAGE_ERROR';
