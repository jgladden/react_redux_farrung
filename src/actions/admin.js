import axios from 'axios';
import * as types from './types';
import { getAdminUrl } from '../config';

export function fetchAdmin(token) {
  const config = {
    headers: {
      'Authorization': `bearer ${token}`
    }
  };
  return dispatch => {
    dispatch(getAdminItems());
    return axios.get(getAdminUrl, config)
      .then(response => dispatch(
        getAdminItemsSuccess(response.data))
      )
      .catch(error => dispatch(
        getAdminItemsError(error.toString()))
      );
  };
}

export const getAdminItems = () => ({
  type: types.GET_ADMIN_ITEMS
});

export const getAdminItemsSuccess = payload => ({
  type: types.GET_ADMIN_ITEMS_SUCCESS,
  payload
});

export const getAdminItemsError = payload => ({
  type: types.GET_ADMIN_ITEMS_ERROR,
  payload
});

export const setDisplayType = payload => ({
  type: types.SET_DISPLAY_TYPE,
  payload
});

export const setDisplayArchived = payload => ({
  type: types.SET_DISPLAY_ARCHIVED,
  payload
});

export const mergeAdminItem = payload => ({
  type: types.MERGE_ADMIN_ITEM,
  payload
});

export const removeAdminItem = payload => ({
  type: types.REMOVE_ADMIN_ITEM,
  payload
});
