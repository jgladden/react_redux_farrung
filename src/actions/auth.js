import axios from 'axios';
import * as types from './types';
import { postAuthUrl } from '../config';

export function submitLogin(authObj) {
  return dispatch => {
    dispatch(postAuth());
    return axios.post(postAuthUrl, authObj)
      .then(response => {
        let error = response.data.error;
        if(!error)
          return dispatch(
            postAuthSuccess(response.data.token)
          );
        return dispatch(
          postAuthError(error)
        );
      })
      .catch(error => dispatch(
        postAuthError(error.toString()))
      );
  };
}

export const postAuth = () => ({
  type: types.POST_AUTH
});

export const postAuthSuccess = payload => ({
  type: types.POST_AUTH_SUCCESS,
  payload
});

export const postAuthError = payload => ({
  type: types.POST_AUTH_ERROR,
  payload
});

export const submitLogout = () => ({
  type: types.AUTH_LOGOUT
});

