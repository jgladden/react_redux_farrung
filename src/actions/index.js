import { jsonFetch } from '../utils/jsonFetch';
import { 
  ADD_PORTFOLIO_TYPE,
  SET_PORTFOLIO_TYPE,
  ADD_PORTFOLIO_ITEM, 
  REMOVE_PORTFOLIO_ITEM 
} from '../constants/action_types';

const api = 'http://farrung.com/api/portfolio1/';

export function fetchPortfolioType(type) {
  let url = api + type;
  return function(dispatch) {
    return jsonFetch(url).then(data => {
      dispatch(addPortfolioType(data));
    });
  };
}

export const addPortfolioType = payload => ({
  type: ADD_PORTFOLIO_TYPE,
  payload
});


export const setPortfolioType = payload => ({
  type: SET_PORTFOLIO_TYPE,
  payload
});

export const addPortfolioItem = payload => ({
  type: ADD_PORTFOLIO_ITEM,
  payload
});

export const removeTodo = payload => ({
  type: REMOVE_PORTFOLIO_ITEM,
  payload
});


