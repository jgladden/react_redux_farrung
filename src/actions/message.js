import axios from 'axios';
import * as types from './types';
import { postMessageUrl } from '../config';

export function submitMessage(messageObj) {
  return dispatch => {
    dispatch(postMessage());
    return axios.post(postMessageUrl, messageObj)
      .then(response => dispatch(
        postMessageSuccess(response.data))
      )
      .catch(error => dispatch(
        postMessageError(error.toString()))
      );
  };
}

export const postMessage = () => ({
  type: types.POST_MESSAGE
});

export const postMessageSuccess = payload => ({
  type: types.POST_MESSAGE_SUCCESS,
  payload
});

export const postMessageError = payload => ({
  type: types.POST_MESSAGE_ERROR,
  payload
});

