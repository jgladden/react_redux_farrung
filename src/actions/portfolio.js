import axios from 'axios';
import * as types from './types';
import { 
  getPortfolioUrl 
} from '../config';

export function fetchPortfolioItems() {
  return dispatch => {
    dispatch(getPortfolioItems());
    return axios.get(getPortfolioUrl)
      .then(response => dispatch(
        getPortfolioItemsSuccess(response.data))
      )
      .catch(error => dispatch(
        getPortfolioItemsError(error.toString()))
      );
  };
}

export const getPortfolioItems = () => ({
  type: types.GET_PORTFOLIO_ITEMS
});

export const getPortfolioItemsSuccess = payload => ({
  type: types.GET_PORTFOLIO_ITEMS_SUCCESS,
  payload
});

export const getPortfolioItemsError = payload => ({
  type: types.GET_PORTFOLIO_ITEMS_ERROR,
  payload
});


export const mergePortfolioItem = payload => ({
  type: types.MERGE_PORTFOLIO_ITEM,
  payload
});

export const removePortfolioItem = payload => ({
  type: types.REMOVE_PORTFOLIO_ITEM,
  payload
});

