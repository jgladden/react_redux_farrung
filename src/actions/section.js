import * as types from './types';

export const setSection = payload => ({
  type: types.SET_SECTION,
  payload
});

