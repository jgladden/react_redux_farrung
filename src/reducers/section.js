import * as types from '../actions/types';

const section = (state = {}, action) => {
  switch (action.type) {
  case types.SET_SECTION: {
    let {
      primary,
      secondary,
      tertiary
    } = action.payload;
    if(primary === 'portfolio')
      secondary = secondary || 'online';
    if(primary === 'admin')
      secondary = secondary || 'list';
    return {  primary, secondary, tertiary }
  }
  default:
    return state;
  }
};

export default section;
