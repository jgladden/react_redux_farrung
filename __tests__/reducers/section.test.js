import section from '../../src/reducers/section';
import { SET_SECTION } from '../../src/actions/types';

describe('section reducers', () => {
  test('initial state is correct', () => {
    const action = { type: '' };
    expect(section(undefined, action)).toEqual({});
  });
  test('set section primary returns expected state', () => {
    const action = { type: SET_SECTION, payload: { primary: 'portfolio' } };
    expect(section(undefined, action)).toMatchSnapshot();
  });
  test('set section secondary returns expected state', () => {
    const action = { type: SET_SECTION, payload: { primary: 'portfolio', secondary: 'print' } };
    expect(section(undefined, action)).toMatchSnapshot();
  });
  test('set section tertiary returns expected state', () => {
    const action = { type: SET_SECTION, payload: { primary: 'portfolio', secondary: 'print', tertiary: '22' } };
    expect(section(undefined, action)).toMatchSnapshot();
  });
});
