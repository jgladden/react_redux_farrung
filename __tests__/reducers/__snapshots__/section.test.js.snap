// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`section reducers set section primary returns expected state 1`] = `
Object {
  "primary": "portfolio",
  "secondary": "online",
  "tertiary": undefined,
}
`;

exports[`section reducers set section secondary returns expected state 1`] = `
Object {
  "primary": "portfolio",
  "secondary": "print",
  "tertiary": undefined,
}
`;

exports[`section reducers set section tertiary returns expected state 1`] = `
Object {
  "primary": "portfolio",
  "secondary": "print",
  "tertiary": "22",
}
`;
